gitlab/resource_gitlab_managed_license_test.go

```diff
@@ -0,0 +1,151 @@
+					testAccCheckGitlabManagedLicenseStatus("gitlab_managed_license.fixme", &managedLicense),
+				),
+			},
+			{
+				SkipFunc:          isRunningInCE,
+				ResourceName:      "gitlab_managed_license.fixme",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckGitlabManagedLicenseStatus(n string, license *gitlab.ManagedLicense) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("not Found: %s", n)
+		}
+
+		licenseStatus := rs.Primary.Attributes["approval_status"]
+		project := rs.Primary.Attributes["project"]
+		if project == "" {
+			return fmt.Errorf("no project ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*gitlab.Client)
+
+		licenses, _, err := conn.ManagedLicenses.ListManagedLicenses(project)
+		if err != nil {
+			return err
+		}
+
+		for _, gotLicense := range licenses {
+			if gotLicense.ApprovalStatus == *stringToApprovalStatus(licenseStatus) {
+				*license = *gotLicense
+				return nil
+			}
+		}
+		return fmt.Errorf("managed license does not exist")
+	}
+}
+
+func testAccCheckManagedLicenseDestroy(state *terraform.State) error {
+	conn := testAccProvider.Meta().(*gitlab.Client)
+
+	for _, rs := range state.RootModule().Resources {
+		if rs.Type != "gitlab_managed_licence" {
+			continue
+		}
+
+		id, _ := strconv.Atoi(rs.Primary.ID)
+		pid := rs.Primary.Attributes["project"]
+
+		license, _, err := conn.ManagedLicenses.GetManagedLicense(pid, id)
+		if err == nil {
+			if license != nil && license.ID == id {
+				return fmt.Errorf("license still exists")
+			}
+		}
+
+		if !is404(err) {
+			return err
+		}
+		return nil
+	}
+	return nil
+}
+
+func testAccCheckGitlabManagedLicenseExists(n string, license *gitlab.ManagedLicense) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not Found: %s", n)
+		}
+
+		licenseName := rs.Primary.Attributes["name"]
+		project := rs.Primary.Attributes["project"]
+		if project == "" {
+			return fmt.Errorf("no project ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*gitlab.Client)
+
+		licenses, _, err := conn.ManagedLicenses.ListManagedLicenses(project)
+		if err != nil {
+			return err
+		}
+
+		for _, gotLicense := range licenses {
+			if gotLicense.Name == licenseName {
+				*license = *gotLicense
+				return nil
+			}
+		}
+		return fmt.Errorf("managed license does not exist")
+	}
+}
+
+func testManagedLicenseConfig(rInt int, status string) string {
+	return fmt.Sprintf(`
+resource "gitlab_project" "foo" {
+  name = "foo-%d"
+  description = "Terraform acceptance tests"
+
+  # So that acceptance tests can be run in a gitlab organization
+  # with no billing
+  visibility_level = "public"
+}
+
+resource "gitlab_managed_license" "fixme" {
+  project = "${gitlab_project.foo.id}"
+  name = "FIXME-%d"
+  approval_status = "%s"
+}
+	`, rInt, rInt, status)
```

Comment:
I believe this is the bug causing the test failure

```suggestion
	`, rInt, status)
```

Reply:
Thanks... I was trying to figure out what's going on with the ee tests, given they all pass on gitlab.com. If I remove that `rInt`, I'll have to remove the %d on either the project or the License. You think it's possibly just too long of a license name? I'll commit that and give it a try. Hard to debug since I can't repro locally :-/

Reply:
oops I thought I only saw two format args. disregard.

Reply:
@armsnyder  - I'm struggling here a bit, open to any ideas you might have on the `403 forbidden` issue. I can see in the makefile the procedure for generating and committing the `GITLAB_TOKEN`, and the scopes that it has when it's generated. When using the exact same scopes on an API token, the tests pass targeting gitlab.com. I checked similar EE tests in other resources, and the only difference I can find between passing and failing is `testAccPreCheck(t)` in the PreCheck function, which only checks to ensure that GITLAB_TOKEN is set within the environment variables.

At this point, I feel silly for even asking this, but... do we know if the license that's provided for EE is maybe a "Premium" license instead of an "Ultimate" license? I ran a grep command to pull down all resources that are checking `isRunningInCE` as their SkipFunc, and it includes the following resources:

- branch protection (Premium)
- managed license (current problem child 😛 )
- group ldap link (All tiers)
- github integration (Premium)
- project test (All tiers)
- group variables (All tiers)
- project level MR approvals (Premium)

So... none of the existing tests would require an ultimate license except for the currently failing test. Again.. I feel a little silly for asking, but I'm not sure what else it could be at this point. The [403 documentations on GitLab's API pages](https://docs.gitlab.com/ee/api/#status-codes) don't seen to provide much info either.

Reply:
@nagyv @nicholasklick Do you know what type of license we use for our CI tests?

I was able to get this `TestAccGitlabManagedLicense_basic` test to pass locally, using an Ultimate license, but it is failing in our CI workflow.

```shell
% curl -s -H 'PRIVATE-TOKEN: ACCTEST1234567890123' pc.local:8080/api/v4/license | jq -r '.plan'
ultimate
```

Ran test on commit bd14ce2374623398b7855a402cfceb789690f89a

```shell
% GITLAB_BASE_URL=http://pc.local:8080/api/v4 make testacc RUN=TestAccGitlabManagedLicense_basic
TF_ACC=1 GITLAB_TOKEN=ACCTEST1234567890123 GITLAB_BASE_URL=http://pc.local:8080/api/v4 go test -v ./gitlab -test.run TestAccGitlabManagedLicense_basic -timeout 40m
=== RUN   TestAccGitlabManagedLicense_basic
--- PASS: TestAccGitlabManagedLicense_basic (185.27s)
PASS
ok  	github.com/gitlabhq/terraform-provider-gitlab/gitlab	198.475s
```

Reply:
I'll ping @nicholasklick internally. I hope he knows.

Reply:
It's premium -- I just checked the /license endpoint via a workflow.

I have an ultimate license which @nicholasklick created for me, which we should be able to use here right?

Reply:
@armsnyder Yes, that should work, I think.

Reply:
@PatrickRice-KSC You won't need to rebase for this one. 😄 

@nagyv Cool. I don't have access to add secrets. Are you on keybase.io?

Reply:
@PatrickRice-KSC Hmm the license was switched out, but the CI is still failing. Maybe try rebasing again? 😁

Reply:
😄  sure - I'm finishing up an MR then I'll rebase there in an hour or so

Reply:
OK, rebased - going to address your PR findings now